rbx_xml/src/types/tags.rs

```diff
@@ -0,0 +1,60 @@
+use std::convert::TryInto;
+use std::io::{Read, Write};
+
+use rbx_dom_weak::types::Tags;
+
+use crate::{
+    core::XmlType,
+    deserializer_core::XmlEventReader,
+    error::{DecodeError, EncodeError},
+    serializer_core::{XmlEventWriter, XmlWriteEvent},
+};
+
+impl XmlType for Tags {
+    const XML_TAG_NAME: &'static str = "BinaryString";
+
+    fn write_xml<W: Write>(&self, writer: &mut XmlEventWriter<W>) -> Result<(), EncodeError> {
+        // FIXME: BinaryString should have an is_empty method.
+        let contents: Vec<u8> = self.into();
+        if !contents.is_empty() {
+            writer.write(XmlWriteEvent::cdata(&base64::encode(&contents)))?;
+        }
+
+        Ok(())
+    }
+
+    fn read_xml<R: Read>(reader: &mut XmlEventReader<R>) -> Result<Tags, DecodeError> {
+        let value = reader.read_base64_characters()?;
+        Ok(value.try_into().unwrap_or_default())
```

Comment:
I wanted to use `map_err` and the question mark operator here but couldn't get it to compile :/